core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -209,7 +219,8 @@ public SelfT withNodeFilter(Predicate<Node> nodeFilter) {
    * <p>Note that this can also be provided by the configuration; if both are defined, this method
    * takes precedence.
    */
-  public SelfT withKeyspace(CqlIdentifier keyspace) {
+  @NonNull
+  public SelfT withKeyspace(@NonNull CqlIdentifier keyspace) {
```

Comment:
Actually our integration tests rely on the keyspace being nullable 😃 
And it makes sense if you're injecting the values from some other part of your program where they can be null. It avoids having to surround every builder call with a null check:
```java
if (keyspace != null) {
  builder = builder.withKeyspace(keyspace);
}
```